### ISSUE TITLE
[Bug]: Cannot select "Select all" option in MultiSelect widget using keyboard navigation

### ISSUE THREAD

**Issue by shadabbuchh** [appsmithorg/appsmith#18304]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

At present we are not able to select the `Select All` option in a MultiSelect widget using keyboard navigation.

https://www.loom.com/share/3e40004a757241c39b24bf96d7bfd9a6

### Steps To Reproduce

1. D&D a MultiSelect widget.
2. Enable the `Enable Select All` property.
3. Now click on the dropdown and use keyboard navigation.

### Version

Cloud

**Comment by dilippitchika** [comment]
Assuming 30% use keyboard navigation

## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 64 |
| Effort (months)  | 0.25 |
